File: 2900. Longest Unequal Adjacent Groups Subsequence I/getLongestSubsequence.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<string> getLongestSubsequence(vector<string>& words, vector<int>& groups) {
        vector<string> ans = {words[0]};
        int lastIndex = 0;

        for(int i=1; i<words.size(); i++){
            if(groups[lastIndex] != groups[i]){
                ans.push_back(words[i]);
                lastIndex = i;
            }
        }

        return ans;
    }
};

File: 2900. Longest Unequal Adjacent Groups Subsequence I/getLongestSubsequence.js
/**
 * @param {string[]} words
 * @param {number[]} groups
 * @return {string[]}
 */
var getLongestSubsequence = function(words, groups) {
    let ans = [words[0]]
    let lastIndex = 0

    for(let i=1 ; i<words.length ;i++){
        if(groups[lastIndex] !== groups[i]){
            ans.push(words[i])
            lastIndex = i
        }
    }

    return ans;
};